### PR TITLE
samples: lwm2m_client: config changes

### PR DESCRIPTION
**Pull request by VeijoPesonen**
Drops the workaround which enabled TLV so that bootstrap would succeed with v1.1

Depends on nrfconnect/sdk-zephyr#832

### CODE REVIEW

**Review by SeppoTakalo — APPROVED** [review]
Looks OK.
Does this PR have to have DoNotMerge flag, because it depends on manifest change?

Also, do we need this now, or should we wait for next upmerge?

File: west.yml [hunk 1]
```diff
@@ -50,7 +50,7 @@ manifest:
     # https://developer.nordicsemi.com/nRF_Connect_SDK/doc/latest/zephyr/guides/modules.html
     - name: zephyr
       repo-path: sdk-zephyr
-      revision: 10da7c808f2c8a30570542d8faa035df9c47816f
+      revision: pull/832/head
```
Comment (rlubos) [inline west.yml:53]: Please update SHA: `814e868e90d0945f9831b73617e4e869d4979739`

Comment (VeijoPesonen) [reply west.yml]: SHA has been updated and a changelog entry has been added.

**Review by rlubos — APPROVED** [review]

File: doc/nrf/releases/release-notes-changelog.rst [hunk 2]
```diff
@@ -173,6 +173,13 @@ nRF9160 samples
 
     * Default configuration conforms to the LwM2M specification v1.0 instead of v1.1.
       For enabling v1.1 there is already an overlay file.
+    * Bootstrap is no more TLV only. With v1.1 preferred content format is send in the bootstrap request.
+      SenML CBOR takes precedence over SenML JSON and OMA TLV, when enabled.
```
Comment (umapraseeda) [inline doc/nrf/releases/release-notes-changelog.rst:177]:
```suggestion
    * Bootstrap is no more TLV only. With v1.1 preferred content format is sent in the bootstrap request.
      SenML CBOR takes precedence over SenML JSON and OMA TLV, when enabled.
```

File: doc/nrf/releases/release-notes-changelog.rst [hunk 3]
```diff
@@ -173,6 +173,13 @@ nRF9160 samples
...
+
+  * Added:
+
+    * CoAP max message size is set to 1280 by default.
+    * Number of SenML CBOR records is set to higher to cope with data exchange after registration with Coiote server.
```
Comment (umapraseeda) [inline doc/nrf/releases/release-notes-changelog.rst:182]:
```suggestion
    * Number of SenML CBOR records is set to a higher value to cope with data exchange after registration with Coiote server.
```

**Comment by VeijoPesonen** [comment]
@umapraseeda I applied your suggestions.

File: doc/nrf/releases/release-notes-changelog.rst [hunk 4]
```diff
@@ -173,6 +173,13 @@ nRF9160 samples
 
     * Default configuration conforms to the LwM2M specification v1.0 instead of v1.1.
       For enabling v1.1 there is already an overlay file.
+    * Bootstrap is no more TLV only. With v1.1 preferred content format is sent in the bootstrap request.
+      SenML CBOR takes precedence over SenML JSON and OMA TLV, when enabled.
```
Comment (umapraseeda) [inline doc/nrf/releases/release-notes-changelog.rst:177]:
```suggestion
    * Bootstrap is not TLV only anymore. 
      With v1.1, preferred content format is sent in the bootstrap request.
      SenML CBOR takes precedence over SenML JSON and OMA TLV, when enabled.
```

**Review by umapraseeda — APPROVED** [review]
just one more minor fix..

**Comment by VeijoPesonen** [comment]
> just one more minor fix..

Done.

File: samples/nrf9160/lwm2m_client/overlay-assist-cell.conf [hunk 5]
```diff
@@ -7,5 +7,4 @@ CONFIG_LWM2M_VERSION_1_1=y
 CONFIG_BASE64=y
```
Comment (jheiskan81) [inline samples/nrf9160/lwm2m_client/overlay-assist-cell.conf:7]: Can we remove CONFIG_BASE64 , CONFIG_LWM2M_RW_SENML_JSON_SUPPORT and   CONFIG_LWM2M_SECONDS_TO_UPDATE_EARLY from those agps and cell assintans overlays?

Comment (VeijoPesonen) [reply samples/nrf9160/lwm2m_client/overlay-assist-cell.conf:7]: Will do that in another PR.

**Review by jheiskan81 — APPROVED** [review]

**Comment by VeijoPesonen** [comment]
Rebased to retrigger the tests. By a quick look the failure in CI was unrelated to my changes.

**Comment by VeijoPesonen** [comment]
There's a bug on the test plan which cause the PR check failure. I'll do rebase to trigger the tests one more time.

**Comment by nvlsianpu** [comment]
@VeijoPesonen You can rebase this - sdk-zephyr reference is already ahead of what this PR introduces,

**Comment by VeijoPesonen** [comment]
> @VeijoPesonen You can rebase this - sdk-zephyr reference is already ahead of what this PR introduces,

Done.

**Pull request merged**